search.go

```diff
@@ -0,0 +1,169 @@
+package opensearchtools
+
+import (
+    "bytes"
+    "context"
+    "encoding/json"
+    "net/http"
+
+    "github.com/opensearch-project/opensearch-go/v2"
+    "github.com/opensearch-project/opensearch-go/v2/opensearchapi"
+
+    "github.com/CrowdStrike/opensearchtools/search"
+)
+
+// SearchRequest wraps the functionality of [opensearchapi.SearchRequest] by supporting request body creation.
+// A simple term query search as an example:
+//
+//	req := NewSearchRequest()
+//	req.SetIndices("example_index")
+//	req.SetQuery(search.NewTermQuery("field", "basic")
+//	results, err := req.Do(context.Background(), client)
+type SearchRequest struct {
+    Query search.Query
+    Index []string
+    Size  *int
+    Sort  []*search.Sort
+}
+
+// NewSearchRequest instantiates an empty SearchRequest.
+// An empty SearchRequest will search across all indices and return the top 10 documents with the default [sorting].
+//
+// [sorting]: https://opensearch.org/docs/latest/opensearch/search/sort/
+func NewSearchRequest() *SearchRequest {
+    return &SearchRequest{}
+}
+
+// Source translates the SearchRequest into the shape expected by OpenSearch.
+func (r *SearchRequest) Source() (any, error) {
```

Comment:
Is this so that `SearchRequest` satisfies the `search.Query` interface?  If so, call that out.

Reply:
It's not. It's just a convenient utility. It also allows consumers to call it and print out the json struct to debug their constructed search requests.